plugins/machina-tools.js

```diff
@@ -0,0 +1,554 @@
+    console.log('conflictAreaLayer disabled');
+    $('.leaflet-control-layers-list span:contains("Machina Conflict Area")').parent('label').addClass('disabled').attr('title', 'Zoom in to show those.');
+  }
+};
+
+machinaTools.guessLevelByRange = function (linkLength) {
+  for (var level = 0; level <= 8; ++level) {
+    if (window.LINK_RANGE_MAC[level] >= linkLength) {
+      return level;
+    }
+  }
+  return 0;
+};
+
+machinaTools.linkAdded = function (data) {
+  data.link.on('add', function () {
+    // if (TEAM_NAMES[this.options.team] != undefined) {
+    if (window.TEAM_NAMES[this.options.team] === window.TEAM_NAME_MAC) {
+      var link = data.link.options.data;
+
+      // add destination portal - 1 level
+      machinaTools.drawExclusion(link.dGuid, 1, machinaTools.getDLatLng(link), true);
+
+      // add origin portal - level based on link length
+      var linkLength = machinaTools.getLinkLength(link);
+      var level = machinaTools.guessLevelByRange(linkLength);
+      machinaTools.drawExclusion(link.oGuid, level, machinaTools.getOLatLng(link), true);
+    }
+    // }
+  });
+};
+
+function humanFileSize(size) {
+  var i = size === 0 ? 0 : Math.floor(Math.log(size) / Math.log(1024));
+  return (size / Math.pow(1024, i)).toFixed(2) * 1 + ' ' + ['B', 'kB', 'MB', 'GB', 'TB'][i];
+}
+
+machinaTools.showConflictAreaInfoDialog = function () {
+  var html = $('<div>');
+  if (machinaTools.conflictAreaLast && machinaTools.conflictAreaLast.getLayers().length > 0) {
+    var ul = $('<ul>');
+    ul.appendTo(html);
+
+    var conflictAreaLayers = machinaTools.conflictAreaLast.getLayers();
+    var conflictLayerGeometry = conflictAreaLayers[0].feature.geometry;
+    ul.append($('<li>', { html: `Cluster area size: ${humanFileSize(sizeof(conflictLayerGeometry))}` }));
+
+    var unitedAreas = 0;
+    var totalPoints = 0;
+
+    if (conflictLayerGeometry.type === 'Polygon') {
+      unitedAreas = 1;
+      totalPoints = conflictLayerGeometry.coordinates[0].length;
+    } else if (conflictLayerGeometry.type === 'MultiPolygon') {
+      unitedAreas = conflictLayerGeometry.coordinates.length;
+      totalPoints = conflictLayerGeometry.coordinates.flatMap((v) => v).flatMap((v) => v).length;
+    }
+    ul.append($('<li>', { html: `Continuous areas: ${unitedAreas}` }));
+    ul.append($('<li>', { html: `Total area points: ${totalPoints}` }));
+  } else {
+    html.append('No machina clusters found');
+  }
+
+  dialog({
+    html: html,
+    title: 'Machina Conflict Area Info',
+    id: 'machina-conflict-area-info',
+    width: 'auto',
+  });
+};
+
+machinaTools.resetConflictArea = function () {
+  delete machinaTools.conflictArea;
+  delete machinaTools.conflictAreaLast;
+  machinaTools.conflictAreaLayer.clearLayers();
+};
+
+function setupLayers() {
+  // This layer is added to the layer chooser, to be toggled on/off, regardless of zoom.
+  machinaTools.displayLayer = new L.LayerGroup([], { minZoom: 15 });
```

Comment:
this gives to the user a feedback in the layer chooser that the layer can not be display
then does this replace the line 409 ?
 ```js
$('.leaflet-control-layers-list span:contains("Machina Level Up Link Radius")').parent('label').removeClass('disabled').attr('title', '');
```

Reply:
yes, but I see this a redundant feedback for the user
not that is important ^^